### ISSUE TITLE
"No capable aircraft available" message when there are aircraft available.

### ISSUE THREAD

**Issue by ColdFrontWI** [dcs-liberation/dcs_liberation#2745]
### Affected versions

6.1.1

### Build information

DCS Liberation 6.1.1

### Description

I am getting a "No capable aircraft available" message when trying to plan a mission, but there are aircraft and pilots available.

In the save file attached, I have:
- 12 F-16s, only 4 assigned to missions
- 7 F-18s, only 4 assigned to missions
- 23 F-15s, 20 assigned to missions

I was able to plan a number of packages, but now any time I try to create a new mission, regardless of task, I get the "No capable aircraft available" message.  I can see the aircraft numbers when I click on the home airfield, but they don't show up when I select Air Wing --> Inventory.  No idle aircraft are shown, only assigned.  Relevant squadrons have plenty of pilots available.  If I remove a flight or reduce the number of aircraft, those become available for tasking elsewhere, but I don't have access to all aircraft.

I have seen this a couple of times, and I've tried to re-create it, but can't seem to figure out how it happens.  In this case, before the above happened, there was a flight plan shown on the map with no package/flight associated with it.  I closed and re-opened, and started getting the "No capable aircraft..." message.

Hopefully I'm not missing anything silly...

Thanks,
Eric

[Turn11.zip](https://github.com/dcs-liberation/dcs_liberation/files/10832571/Turn11.zip)

### Save game and other files

No error message displayed.

**Comment by Starfire13** [comment]
This is a very longstanding bug that affects those who play on a dedicated server. Just pass turn to get your planes back.

**Comment by Starfire13** [comment]
This is a duplicate of #1880 but that bug report was closed some time back due to lack of replicability. As far as I can tell, pretty much everyone who encounters this bug is playing on a dedicated server (whether running on the same machine or a different one), so it's not an issue I encounter (since I don't play on a dedicated server). However, while it appears to have happened to a great many folks, it does not occur on every single turn. Potentially a bug in the state.json manual submit code, maybe?

**Comment by DanAlbert** [comment]
FWIW, manual submit is just about the only workflow I use when developing, and when I do actually play I use a dedicated server (on the same machine) and I don't recall having ever seen this :(

**Comment by Starfire13** [comment]
Ah, that's frustrating. The bug clearly exists considering how many people have encountered it, but I just have no idea why it's happening.

In the attached file, autoplanner is turned on. I'm wondering if the issue might be occuring because the autoplanner has tried to task those aircraft and failed to do so for some reason, and the aircraft are not released back into the available pool when that happens. I've noticed on previous occasions (with bugs that have since been dealt with) that when planning a mission fails, the aircraft selected for that mission go missing till the turn is passed.

**Comment by ColdFrontWI** [comment]
Thanks.  I found the earlier post after I put mine up.  For what it's worth, this was *not* a dedicated server / manual submission.  I ran the mission single-player and accepted the auto-generated results after the mission ended in DCS.

In case it helps, I'm attaching a zip with bunch more files in case anyone wants to dig.  Here's what those files are with a bit about how they were created as well as I can reconstruct:
- **Turn10.liberation** and **Turn10.miz:** Liberation file from the prior turn as well as the mission file I flew.
- **Turn10_state.json:** The state.json file that was generated post Turn10.
- **Turn10.zip.acmi:** *(separate .zip file)* Tacview file from Turn 10.  One oddity: According to Tacview my tanker was destroyed 8:24 into the mission as it was returning to base WAY before schedule, HOWEVER, I still have a tanker available in Turn 11.
- **Turn11.liberation** and **Turn11.miz:** Liberation file from the current turn where the issue exists and a mission file, not yet flown.  Here's the sequence for creation of this file:
    - Click "Take Off" from within Turn 10.
    - Modify Turn10.miz in the mission editor to move my flight's parking.  Not the first time I've done this.
    - Fly Turn10.miz.
    - Accept the *auto-submitted* results.
    - Save "Turn11.liberation" for the first time.
    - Delete/modify many the auto-planned missions.
    - Notice a weird case where there is a flight plan line on the map with NO packages / aircraft assigned to it.  I THINK this arose when I started to plan a package, but then closed it when there were no flights in place, but I've tried to re-create this with no luck.
    - At this point tasked/available aircraft are as follows:
        - 4/4 A-10s
        - **4/7** F-18s
        - **20/23** F-15s
        - **4/12** F-16s
        - 1/1 E-3
        - 1/1 KC-135
    - The remaining aircraft (3x F-18s, 3x F-15s, and 8x F-16s) show up when clicking on the airbases, but not in the air wing inventory, and I can't plan them.
- **liberation_Turn11 error.log:** This is a log file that was captured at a point when the game was in the error state.  A few key times:
    - **2023-02-25 14:41:45,915:** Turn10.miz is generated
    - **2023-02-25 16:02:21,804:** "Committing mission results."
    - **2023-02-25 16:02:22,517:** Looks like blue auto-planning is done.
    - **2023-02-25 16:03:06,426:** When I would have first saved Turn11.liberation.
    - From there I was manually planning.  Not sure where along the way the error arose.
- **Turn11_manual resubmit.liberation** and **.miz**: Steps to create these files:
    - Open "Turn10.liberation"
    - Click "Take Off"
    - Click "Manually Submit"
    - Submit "Turn10_state.json"
    - At this point, based on the auto-generated missions, all aircraft are available.  I can see idle aircraft in the inventory:
        - 3x F-18s (4/7 planned)
        - 1x F-15 (22/23 planned)
        - All of the other aircraft are tasked (4x A-10s, 12x F-16s, 1x E-3, and 1x KC-135)
    - Generate the .miz file.

Thanks for your help, and thanks for all your work creating Liberation.  I'm really loving it...makes the game a LOT more engaging.
Eric

[No Capable Aircraft bug 2023-02-26_v02.zip](https://github.com/dcs-liberation/dcs_liberation/files/10834923/No.Capable.Aircraft.bug.2023-02-26_v02.zip)


[Turn10_Tacview.zip](https://github.com/dcs-liberation/dcs_liberation/files/10834926/Turn10_Tacview.zip)

**Comment by Starfire13** [comment]
Can you describe what you're doing with "Delete/modify many the auto-planned missions", and why you do that in the mission editor rather than in Liberation? I typically do all that stuff in Liberation before the mission is generated, and typically the only thing I change in the mission editor is the weather.

**Comment by ColdFrontWI** [comment]
Sorry for the confusion...the "Delete/modify" part is done in Liberation.  I have auto-planner turned on, but I usually make a lot of adjustments to it.  Typically I'll add extra flights to BARCAPs (second pair of F-15s in each time slot), remove strikes on economic targets, refocus SEAD/DEAD based on threat, etc.

I don't usually use the ME at all.  In this case, I just used it to change my parking location.  I'm based at Damascus for the moment and depending on runway in use, the taxi can be LONG.

**Comment by DanAlbert** [comment]
Apologies for taking so long to get to this. I had some down time and once I was back I got caught up in a flurry of DCS features we needed to deal with :(

Has anyone seen this more recently, or has this maybe been fixed (unlikely)? If anyone has fresh artifacts (save game and state.json) that'd make this a bit easier. If not, nbd, I can debug this in 6.1.1, it's just more work to do that so I thought I'd check first.
